modules/rostests/apitests/ntdll/load_notifications/load_notifications.c

```diff
@@ -0,0 +1,70 @@
+#define WIN32_NO_STATUS
+#define _INC_WINDOWS
+#define COM_NO_WINDOWS_H
+
+#include <windows.h>
+
+
+#if defined(_MSC_VER)
+#define _CRTALLOC(x) __declspec(allocate(x))
+#elif defined(__GNUC__)
+#define _CRTALLOC(x) __attribute__ ((section (x) ))
+#else
+#error Your compiler is not supported.
+#endif
+
+
+static VOID (WINAPI* pTlsCallback)(IN HINSTANCE hDllHandle, IN DWORD dwReason, IN LPVOID lpvReserved);
+
+VOID WINAPI
+TlsCallback(IN HANDLE hDllHandle,
+            IN DWORD dwReason,
+            IN LPVOID lpvReserved)
+{
+    if (!pTlsCallback)
+        pTlsCallback = (VOID*)GetProcAddress(NULL, "notify_TlsCallback");
+    if (pTlsCallback)
+    {
+        pTlsCallback(hDllHandle, dwReason, lpvReserved);
+        return;
+    }
+    OutputDebugStringA("WARNING: load_notifications.dll loaded from a process without notify_TlsCallback\n");
+}
+
+#if defined(_MSC_VER)
+#pragma section(".rdata$T",long,read)
+#pragma section(".tls",long,read,write)
+#pragma section(".tls$ZZZ",long,read,write)
+#endif
+
+_CRTALLOC(".tls") char _tls_start = 0;
+_CRTALLOC(".tls$ZZZ") char _tls_end = 0;
+
+PIMAGE_TLS_CALLBACK __xl_a[2] = { &TlsCallback, NULL };
+
+ULONG _tls_index = 0;
+
+_CRTALLOC(".rdata$T") const IMAGE_TLS_DIRECTORY _tls_used = {
+  (ULONG_PTR) &_tls_start+1, (ULONG_PTR) &_tls_end,
+  (ULONG_PTR) &_tls_index, (ULONG_PTR) (__xl_a),
+  (ULONG) 0, (ULONG) 0
```

Comment:
Any particular reason this needs casts?

Reply:
Copied straight from https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/lib/crt/startup/tlssup.c;hb=74adcadbd08f782cc43017b12378611bbff50836#l57
I'll indicate this with a comment.
Only `__xl_a` is different, because I could use the array as-is, and this file needs other people to be able to insert items.